### PR TITLE
MRG: Enforce ints for events

### LINKED ISSUE #2406: EpochsArray should throw clearer error than 'no matching events'
I think there may be a numerical error when creating epochs with `EpochsArray`? Or is it a deliberate choice ?

```
# prepare
import numpy as np
from mne.epochs import EpochsArray
from mne.io.meas_info import create_info
ntrial, nchan, ntime = 100, 20, 10
data = np.ones((ntrial, nchan, ntime))
chan_types = ['eeg'] * nchan
chan_names = ['eeg_%02i' % chan for chan in range(nchan)]
events = np.zeros((ntrial, 3))
np.random.seed(0)
events[:, 2] = np.random.randn(1, ntrial)
info = create_info(chan_names, 256, chan_types)
```
```
# this works
events[:, 2] = np.round(1e7 * np.random.randn(1, ntrial))
EpochsArray(data, info, events=events, verbose=False)
```
```
# this doesn't
events[:, 2] = np.random.randn(1, ntrial)
EpochsArray(data, info, events=events, verbose=False)
```
```
> ValueError: No matching events found for 1.02017271172 (event id 1)
```

### PR DESCRIPTION
**Pull request by Eric89GXL**
Fixes the six bugs we had not making `events` integers. Builds on #2411. Closes #2406.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
I also included a few other small fixes for recent `scipy`, and to squash warnings during tests, since I needed those to get things passing again.

File: mne/preprocessing/ecg.py [hunk 1]
```diff
@@ -186,8 +186,9 @@ def find_ecg_events(raw, event_id=999, ch_name=None, tstart=0.0,
     logger.info("Number of ECG events detected : %d (average pulse %d / "
                 "min.)" % (n_events, average_pulse))
 
-    ecg_events = np.c_[ecg_events + raw.first_samp, np.zeros(n_events),
-                       event_id * np.ones(n_events)]
+    ecg_events = np.array([ecg_events + raw.first_samp,
+                           np.zeros(n_events, int),
+                           event_id * np.ones(n_events, int)]).T
```
Comment (agramfort) [inline mne/preprocessing/ecg.py]: np.c_ and np.r_ is not recommended anymore?

**Comment by agramfort** [comment]
thanks heaps @Eric89GXL !

File: mne/preprocessing/ecg.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/preprocessing/ecg.py]: AFAIK they were never recommended for production code, just as shortcuts in
interactive programming. At least I think I read that somewhere.

**Pull request merged**